cocoa/src/appkit.rs

```diff
@@ -4017,6 +4018,66 @@ impl NSToolbar for id {
     }
 }
 
+pub trait NSSpellChecker : Sized {
+    unsafe fn sharedSpellChecker(_: Self) -> id;
+    unsafe fn checkSpellingOfString_startingAt(self,
+                                               stringToCheck: id,
+                                               startingOffset: NSInteger) -> NSRange;
+    unsafe fn checkSpellingOfString_startingAt_language_wrap_inSpellDocumentWithTag_wordCount(
+        self,
+        stringToCheck: id,
+        startingOffset: NSInteger,
+        language: id,
+        wrapFlag: BOOL,
+        tag: NSInteger) -> (NSRange, NSInteger);
+    unsafe fn uniqueSpellDocumentTag(_: Self) -> NSInteger;
+    unsafe fn closeSpellDocumentWithTag(self, tag: NSInteger);
+    unsafe fn ignoreWord_inSpellDocumentWithTag(self, wordToIgnore: id, tag: NSInteger);
+}
+
+impl NSSpellChecker for id {
+    unsafe fn sharedSpellChecker(_: Self) -> id {
+        msg_send![class!(NSSpellChecker), sharedSpellChecker]
+    }
+
+    unsafe fn checkSpellingOfString_startingAt(self,
+                                               stringToCheck: id,
+                                               startingOffset: NSInteger) -> NSRange {
+        msg_send![self, checkSpellingOfString:stringToCheck startingAt:startingOffset]
+    }
+
+    unsafe fn checkSpellingOfString_startingAt_language_wrap_inSpellDocumentWithTag_wordCount(
+        self,
+        stringToCheck: id,
+        startingOffset: NSInteger,
+        language: id,
+        wrapFlag: BOOL,
+        tag: NSInteger) -> (NSRange, NSInteger) {
+        let wordCount = ptr::null_mut();
```

Comment:
This won't do the right thing. We should initialize `let mut wordCount = 0;`, pass a mutable reference as the argument, and return the value of `wordCount`.

Reply:
Whoops, fixed!